app/src/main/java/com/android/shelfLife/viewmodel/InvitationViewModel.kt

```diff
@@ -1,29 +1,79 @@
 package com.android.shelfLife.viewmodel
 
+import android.util.Log
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.android.shelfLife.model.newInvitations.Invitation
 import com.android.shelfLife.model.newInvitations.InvitationRepository
 import com.android.shelfLife.model.user.UserRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
 
 @HiltViewModel
 class InvitationViewModel
 @Inject
 constructor(
-    private val invitationRepository: InvitationRepository,
-    private val userRepo: UserRepository
+  private val invitationRepository: InvitationRepository,
+  private val userRepo: UserRepository
 ) : ViewModel() {
 
-  val invitations = invitationRepository.invitations
+  private val _invitations = MutableStateFlow<List<Invitation>>(emptyList())
+  val invitations: StateFlow<List<Invitation>> = _invitations
+
+  init {
+    viewModelScope.launch {
+      val invitationList = mutableListOf<Invitation>()
+      userRepo.invitations.collect { invitationUIDs ->
+        invitationList.clear()
+        invitationUIDs.forEach { uid ->
+          try {
+            val invitation = invitationRepository.getInvitation(uid)
```

Comment:
we should be calling a function called getInvitations that you deleted in this pr instead of iterating over all the list and getting them one by one. this is to keep consistency.